File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maj_ind=0
        cnt=1
        for i in range(1,len(nums)):
            if nums[i]==nums[maj_ind]:
                cnt+=1
            else:
                cnt-=1
            if cnt==0:
                maj_ind=i;
                cnt=1
        return nums[maj_ind]
        

File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int pos=-1, val_cnt=0;
        for(int i=0;i<nums.size();i++)
        {
            if(val_cnt==0){pos=i; val_cnt++;}
            else{
                if(nums[pos]==nums[i])
                    val_cnt++;
                else val_cnt--;
            }
        }
        return nums[pos];
    }
};